pkg/bytecode/value.go

```diff
@@ -59,3 +60,37 @@ func (s stringVal) Equals(v value) bool {
 	}
 	return s == s2
 }
+
+type arrayVal struct {
+	Elements []value
+}
+
+func (a *arrayVal) Type() *parser.Type {
+	return parser.GENERIC_ARRAY
+}
```

Comment:
This doesn't feel correct, and in fact we only use Value#Type() in panic error messages (where we could just print the name of the go type). Is this usage acceptable, or would it be better to drop Type entirely and avoid the issue?

Reply:
FWIW, types were the trickiest bit for me so far, in parser and evaluator. Maybe its because I didn't model them well, maybe they just are hard, but I think Cam is right you will probably not get past tracking types.
Comment sounds fine for now to me.